NmraDcc.cpp

```diff
@@ -174,9 +175,22 @@
         #define MODE_TP3 pinMode( D7,OUTPUT ) ; // GPIO 13
         #define SET_TP3  GPOS = (1 << D7);
         #define CLR_TP3  GPOC = (1 << D7);
-        #define MODE_TP4 pinMode( D7,OUTPUT ); // GPIO 15
+        #define MODE_TP4 pinMode( D8,OUTPUT ) ; // GPIO 15

```

Comment:
This is a change to non-ESP32 code - is it deliberate or an error?

Reply:
Indeed, this fixes an error in ESP8266 Code. Thanks to Trusty77. 
It has no effect in the release version, because these defines are only for debugging of the bit decoding with a logic analyser.

Reply:
I think that the @chrisridd comment is about how the proposed change to the code is an ESP32-specific change, but the file being changed, in an ideal world, ought not to have system-specific code.  That means that the proposed change has an effect on non-ESP32 library users.

I _do not_ understand what the big-picture plan is for isolating system-specific code from the more-generic code within this set of libraries.  So I cannot suggest a particular strategy for solving the "problem" this change introduces, if there actually is a "problem".  

@kiwi64ajs, can you provide some guidance?

Reply:
A lot was achieved by using the hardware interrupt and system micros but we're always going to have platform specific code that has to live somewhere, so if may as well live behind #if's.

Reply:
Actually @devel-bobm I was just wondering if the change was a stray one or not, and wasn't a comment about the platform specific code in this file.

Thanks for the fix @Trusty77 !

Reply:
Hi all. This change has been proposed in the same pull request because it seems not correct. I was wondering how to implement these debug facilities in ESP32 mode, and found this weird line. I fix it, but dont do the adaptation for ESP32 because of my poor knowledge of its capabilities. All i can say is that with these changes, the ESP32 works, without its debug mode, of course.